test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -3,220 +3,90 @@
-import io.quarkus.runner.RuntimeRunner;
-import io.quarkus.runner.TransformerTarget;
-import io.quarkus.runtime.LaunchMode;
-import io.quarkus.test.common.NativeImageLauncher;
+import io.quarkus.deployment.proxy.ProxyConfiguration;
+import io.quarkus.deployment.proxy.ProxyFactory;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.runner.bootstrap.AugmentAction;
+import io.quarkus.runner.bootstrap.RunningQuarkusApplication;
+import io.quarkus.test.common.DefineClassVisibleClassLoader;
 import io.quarkus.test.common.PathTestHelper;
 import io.quarkus.test.common.PropertyTestUtil;
 import io.quarkus.test.common.RestAssuredURLManager;
-import io.quarkus.test.common.TestInjectionManager;
-import io.quarkus.test.common.TestInstantiator;
 import io.quarkus.test.common.TestResourceManager;
 import io.quarkus.test.common.TestScopeManager;
 import io.quarkus.test.common.http.TestHTTPResourceManager;
 
+//todo: share common core with QuarkusUnitTest
 public class QuarkusTestExtension
-        implements BeforeEachCallback, AfterEachCallback, TestInstanceFactory, BeforeAllCallback {
+        implements BeforeEachCallback, AfterEachCallback, TestInstanceFactory, BeforeAllCallback, InvocationInterceptor,
+        AfterAllCallback {
 
-    private URLClassLoader appCl;
-    private ClassLoader originalCl;
     private static boolean failedBoot;
 
-    /**
-     * As part of the test run we need to create files in the test-classes directory
-     *
-     * We attempt to clean these up with a shutdown hook, but if the processes is killed (e.g. hitting the red
-     * IDE button) it can leave these files behind which interfere with subsequent runs.
-     *
-     * To fix this we create a file that contains the names of all the files we have created, and at the start of a new
-     * run we remove them if this file exists.
-     */
-    private static final String CREATED_FILES = "CREATED_FILES.txt";
-    private final RestAssuredURLManager restAssuredURLManager = new RestAssuredURLManager(false);
+    private static Class<?> actualTestClass;
+    private static Object actualTestInstance;
+    private static ClassLoader originalCl;
+    private static RunningQuarkusApplication runningQuarkusApplication;
+    private static Path testClassLocation;
+    private static boolean allowPackagePrivateMethods;
 
     private ExtensionState doJavaStart(ExtensionContext context, TestResourceManager testResourceManager) {
 
-        final LinkedBlockingDeque<Runnable> shutdownTasks = new LinkedBlockingDeque<>();
+        try {
+            final LinkedBlockingDeque<Runnable> shutdownTasks = new LinkedBlockingDeque<>();
 
-        Path appClassLocation = getAppClassLocation(context.getRequiredTestClass());
+            Path appClassLocation = getAppClassLocation(context.getRequiredTestClass());
 
-        appCl = createQuarkusBuildClassLoader(appClassLocation);
-        originalCl = setCCL(appCl);
+            QuarkusBootstrap.builder(appClassLocation);
```

Comment:
Not that it should be an issue, but I assume this is not needed since `QuarkusBootstrap.builder(appClassLocation).setMode(QuarkusBootstrap.Mode.TEST);` is being used later